lib/bullet/ext/string.rb

```diff
@@ -4,8 +4,13 @@ module Bullet
   module Ext
     module String
       refine ::String do
+        attr_reader :bullet_class_name
+
         def bullet_class_name
-          sub(/:[^:]*?$/, '')
+          @bullet_class_name ||= begin
+            last_colon = self.rindex(':')
```

Comment:
microbenchmark on sub showed rindex is about twice as fast for this sort of work.
memoizing here is just the cherry on top